File: array/1375_bulb_switcher_iii/1375_bulb_switcher_iii.py

class Solution(object):
    def numTimesAllBlue(self, light):
        res = 0
        max_idx = 0
        for i,b in enumerate(light):
            max_idx = max(max_idx, b)
            if i + 1 == max_idx:
                res += 1
        return res


light =[2,1,3,5,4]
res = Solution().numTimesAllBlue(light)
print(res)

File: array/1375_bulb_switcher_iii/1375_bulb_switcher_iii.go
package main

import "fmt"

func numTimesAllBlue(light []int) int {
	var (
		i       int
		rst     int
		maxOpen = 0
		length  = len(light)
	)

	for i = 0; i < length; i++ {
		if light[i] > maxOpen {
			maxOpen = light[i]
		}
		if maxOpen == i+1 {
			rst++
		}
	}
	return rst
}

func main() {
	light := []int{2, 1, 3, 5, 4}
	res := numTimesAllBlue(light)
	fmt.Println(res)
}
